### ISSUE TITLE
Typings issue with v6: Type '{ id: number; } & Reference<User>' is not assignable to type 'ScalarReference<any> | ({ id: any; _id: any; uuid: any; } & Reference<any>)'

### ISSUE THREAD

**Issue by darkbasic** [mikro-orm/mikro-orm#4907]
**Describe the bug**
With the nearing release of account.js 1.0 I've updated the yet-to-be-released [mikro-orm adapter](https://github.com/accounts-js/accounts/tree/master/packages/database-mikro-orm) to v6, but I'm having issues with the typings that I cannot figure out.
Here is a repro: https://github.com/darkbasic/accounts-mikro-orm-v6-repro

**Stack trace**
`src/example/index.ts Ln 20, Col 56`
```
Type 'Email' does not satisfy the constraint 'Email<any>'.ts(2344)
```

`src/example/entities/user.ts Ln 19, Col 3`
```
Type 'Email' does not satisfy the constraint 'Email<any>'.
  Types of property 'user' are incompatible.
    Type '{ id: number; } & Reference<User>' is not assignable to type 'ScalarReference<any> | ({ id: any; _id: any; uuid: any; } & Reference<any>)'.ts(2344)
```

`src/example/mikro-orm-config.ts Ln 28, Col 35`
```
Types of construct signatures are incompatible.
  Type 'new ({ fullName, ...otherProps }: EmailCtorArgs) => Email' is not assignable to type 'new (args: EmailCtorArgs<any>) => Email<any>'.
    Type 'Email' is not assignable to type 'Email<any>'.ts(2419)
User.ts(99, 3): The expected type comes from property 'EmailEntity' which is declared here on type '{ AccountsUser: UserCtor<any, any, any, any>; EmailEntity?: EmailCtor<any> | undefined; SessionEntity?: SessionCtor<any> | undefined; ServiceEntity?: ServiceCtor<...> | undefined; abstract?: boolean | undefined; }'
```

**To Reproduce**
1. `git clone https://github.com/darkbasic/accounts-mikro-orm-v6-repro`
2. `cd accounts-mikro-orm-v6-repro`
3. `yarn`
4. Open the files in an editor or try to compile the code.

**Expected behavior**
There shouldn't be any difference between v5 and v6, especially since they share the same version of Typescript.

**Additional context**
Latest commit of the repro is v6, while [the commit before](https://github.com/darkbasic/accounts-mikro-orm-v6-repro/commit/0481e42c8541b29b03a450eda89c399611b7031b) is v5 and works well.

**Versions**

| Dependency | Version |
| - | - |
| node | v18.18.0 |
| typescript | 5.2.2 |
| mikro-orm | 6.0.0-rc.0 |
| your-driver | postgresql |

**Comment by B4nan** [comment]
Hmm that itself didn't help, the PK also causes issues. But that is probably a wontfix. You could still do this to make it work correctly:

```ts
export class Email<CustomUser extends IUser<any, any, any>> {
  id!: number;
  user!: Reference<CustomUser> & { id: number };
  // ...
}
```

The constructor assignment can be fixed by using `this.user = ref(user);`

**Comment by darkbasic** [comment]
Thanks. I've managed to get it working with rc0 by specifying the bare minimum of types: https://github.com/accounts-js/accounts/commit/d93097619a0ddecd948462cb62d3cdea1917d8b1#diff-70fb6c72f8d04156077fe444cc0ad30f670b74824fce60874a4205c99ad24411

> the PK also causes issues.

Yeah I've encountered that as well.

> You could still do this to make it work correctly

I'm planning to further generalize the code later on in order to support all possible databases including NoSQL, so I'm not sure how that would play with different primary keys.
